src/Rector/Convert/HookConvertRector.php

```diff
@@ -241,17 +243,19 @@ protected function createMethodFromFunction(Function_ $node): ?ClassMethod
         if ($info = $this->getHookAndModuleName($node)) {
             ['hook' => $hook, 'module' => $implementsModule] = $info;
             $procOnly = [
-                'install',
-                'requirements',
-                'schema',
-                'uninstall',
-                'update_last_removed',
-                'module_implements_alter',
-                'hook_info',
-                'install_tasks',
-                'install_tasks_alter',
+              'hook_info',
+              'install',
+              'install_tasks',
+              'install_tasks_alter',
+              'module_implements_alter',
+              'removed_post_updates',
+              'requirements',
+              'schema',
+              'uninstall',
+              'update_dependencies',
+              'update_last_removed',
             ];
-            if (in_array($hook, $procOnly) || str_starts_with($hook, 'preprocess') || str_starts_with($hook, 'process')) {
+            if (in_array($hook, $procOnly)) {
```

Comment:
` || preg_match('/^(post_update_|update_\d+$)/', $hook)` is missing in line 258. see `checkForProceduralOnlyHooks` in https://git.drupalcode.org/project/drupal/-/blob/11.2.x/core/lib/Drupal/Core/Hook/HookCollectorPass.php?ref_type=heads#L585